moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp

```diff
@@ -339,6 +339,25 @@ bool planning_pipeline::PlanningPipeline::generatePlan(const planning_scene::Pla
     display_path_publisher_.publish(disp);
   }
 
+  if (!solved)
+  {
+    // This should alert the user if planning failed because of contradicting constraints.
+    // Could be checked more thoroughly, but it is probably not worth going to that length.
+    bool stacked_constraints = false;
+    if (req.goal_constraints.size() > 1 || req.path_constraints.position_constraints.size() > 1 ||
```

Comment:
Several goal_constraints are perfectly fine. In this fashion you can define several independent goal regions.

Reply:
Yes. This warning is displayed upon planning failure although some users may have valid reasons for setting multiple constraints, but that's why the tone of the error message is "Hey maybe this is what made your planning fail, did you mean to do this?"

What is the change you would like to see in this snippet? I think users advanced enough to set multiple goal regions will have the confidence to ignore the message and not be bothered.

Reply:
The condition should read:
`if (req.path_constraints.position_constraints.size() > 1 || req.path_constraints.orientation_constraints.size() > 1)`
i.e. remove the check `req.goal_constraints.size() > 1`, which is required to define multiple goal regions.
Further below you check each of these goal constraints for multiple position/orientation constraints...

Reply:
I think that might not catch the problem in my ROS Answers post, where I accidentally defined two goals although I only had one robot arm. I would err on the side of displaying too much when things fail. Is there an easy way to make a more precise check? Is it worth the effort to code and execute it?

Reply:
> @henningkayser Why did you merged this without a second review?

@rhaschke Sorry I was a bit too fast here!
It's true that multiple goal constraints should be treated as alternative targets and don't really apply here. Nevertheless, the warning is quite generic and would help in any case where constraints are somehow malformed.

@felixvd did you attempt to fix [your issue](https://answers.ros.org/question/299979/moveit-plan-fails-after-attaching-an-object-but-not-from-rviz-or-after-restarting-requesting-node/) directly?

Reply:
I agree that the warning is generic. However, we shouldn't issue it, when there is no reason for it.
I don't ask for a more thorough implementation. I'm just asking to remove the unnecessary check `req.goal_constraints.size() > 1`.

Reply:
So all goal_constraints are treated as alternatives, so that only one of them has to be fulfilled for the planning to succeed? In that case it should be edited out, I agree. Please confirm first though.

Reply:
Yes, that's correct. I'm not sure though, how this relates to your [original issue](https://answers.ros.org/question/299979/moveit-plan-fails-after-attaching-an-object-but-not-from-rviz-or-after-restarting-requesting-node/). Maybe you can try to reproduce your original issue? Maybe the root cause was something else?

Reply:
The cause was surely that a pose constraint was set for two different links on the same robot because I changed the end effector without clearing the targets. Our end effector had multiple tool tips.

I'll put in another PR for the correction.